pkg/template/factory.go

```diff
@@ -14,20 +16,22 @@ import (
 type Factory struct {
 	coreClient     kubernetes.Interface
 	fetchFactory   fetch.Factory
+	deployFactory  deploy.Factory
 	kbldAllowBuild bool
 	cmdRunner      exec.CmdRunner
+	log            logr.Logger
 }
 
 // NewFactory returns template factory.
-func NewFactory(coreClient kubernetes.Interface, fetchFactory fetch.Factory,
-	kbldAllowBuild bool, cmdRunner exec.CmdRunner) Factory {
+func NewFactory(coreClient kubernetes.Interface, fetchFactory fetch.Factory, deployFactory deploy.Factory,
+	kbldAllowBuild bool, cmdRunner exec.CmdRunner, log logr.Logger) Factory {
 
-	return Factory{coreClient, fetchFactory, kbldAllowBuild, cmdRunner}
+	return Factory{coreClient, fetchFactory, deployFactory, kbldAllowBuild, cmdRunner, log}
 }
 
 // NewYtt returns ytt template.
 func (f Factory) NewYtt(opts v1alpha1.AppTemplateYtt, appContext AppContext) *Ytt {
-	return NewYtt(opts, appContext, f.coreClient, f.fetchFactory, f.cmdRunner)
+	return NewYtt(opts, appContext, f.coreClient, f.fetchFactory, f.deployFactory, f.cmdRunner, f.log)
```

Comment:
only ytt actually implements the k8s version stuff right now... so ignore the other's, threading the deployfactory through them all was just to get them to compile